### DISCUSSION TITLE
How to change the size of an pyLDAvis html file rendered in wave

### DISCUSSION THREAD

**Discussion by gbethardy-h2oai** [h2oai/wave#1373] [Q&A]
This may not be enough info.  I'm generating an html file using pyLDAvis and rendering it in wave.  It doesn't quite fit in the panel, so scroll bars show up.  Is there a way to resize this visualization to fit better in wave?

**Comment by mturoci** [comment]
Can you please post a screenshot? However, I am afraid there is nothing you can do on Wave side. Wave just gives you the container/space to put your HTML into. Ideally, the HTML should be responsive and adhere to the container, but if that's not the case and it has fixed dimensions set, you can try to size the container (frame/markup card) to the same dimensions as the HTML. The screenshot would be good and min repro (even with hardcoded HTML) would be perfect to provide less generic advice though.

**Comment by gbethardy-h2oai** [comment]
I tried resizing the frame/card.   Here are the screen shots.  I can't get
rid of the horizontal scroll bar at the bottom, even if I maximize the
browser to take the entire screen.  I also looked in the pyLDAvis docs for
a way, but didn't find anything.

On Fri, Apr 22, 2022 at 3:57 AM mturoci ***@***.***> wrote:

> Can you please post a screenshot? However, I am afraid there is nothing
> you can do on Wave side. Wave just gives you the container / space to put
> your HTML into. Ideally, the HTML should be responsive and adhere to the
> container, but if that's not the case and it has fixed dimensions set, you
> can try to size the container (frame/markup card) to the same dimensions as
> the HTML. Screenshot would be good and min repro (even with hardcoded HTML)
> would be perfect to less generic advice though.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/h2oai/wave/discussions/1373#discussioncomment-2613925>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXRVHE6AZV63LOSHFS7PUFDVGJLW7ANCNFSM5UA2VRKQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

<https://www.h2o.ai/>

Greg Bethardy, Ph.D.

Customer Data Scientist

***@***.*** | (312) 507-9754

<https://twitter.com/h2oai> <https://www.linkedin.com/company/h2oai>

Democratize AI with the H2O AI Cloud <https://www.h2o.ai/ai-cloud/>

**Comment by gbethardy-h2oai** [comment]
![Screen Shot 2022-04-24 at 4 15 06 PM](https://user-images.githubusercontent.com/98784147/164995274-43e9183f-9f0a-40ca-9c20-7637bfc52b7d.png)

![Screen Shot 2022-04-24 at 4 15 23 PM](https://user-images.githubusercontent.com/98784147/164995278-70756b91-192e-4dfc-b371-f3fcdd29add0.png)

**Comment by mturoci** [comment]
I looked at some pyLDAvis examples and they are hardcoded, not responsive. They use a single SVG for plots with hardcoded width/height instead of viewbox so it's not easily scalable. So your only option is to increase the total width of the card:

* Based on your screenshots, there is still some usable whitespace at the left and right screen edges (I assume you use max-width for the layout).
* Putting the navigation into top header will save you some more space which should be enough to get rid of the scrollbar.

Note that for smaller screens, the scrollbars will still be visible.
